community/kernel/src/main/java/org/neo4j/kernel/impl/core/InMemoryTokenCache.java

```diff
@@ -59,18 +60,18 @@ public void putAll( List<TOKEN> tokens ) throws NonUniqueTokenException
 
         for ( TOKEN token : tokens )
         {
-            putAndEnsureUnique( newNameToId, token, tokenType );
             newIdToToken.put( token.id(), token );
+            putAndEnsureUnique( newNameToId, token, tokenType );
```

Comment:
If we get a duplicate, then this will modify the `idToToken` map by removing the existing value that conflicted with the duplicate, and then throw an exception before adding any of the new values. That's not the behaviour I would expect from a `putAll` method.